hopsworks-common/src/main/java/io/hops/hopsworks/common/featurestore/storageconnectors/s3/FeaturestoreS3ConnectorController.java

```diff
@@ -249,18 +301,48 @@ private void verifyS3ConnectorSecretKey(String secretKey) throws FeaturestoreExc
    */
   private void verifyUserInput(Featurestore featurestore, FeaturestoreS3ConnectorDTO featurestoreS3ConnectorDTO)
     throws FeaturestoreException {
-
     if (featurestoreS3ConnectorDTO == null) {
       throw new IllegalArgumentException("Null input data");
     }
     verifyFeaturestore(featurestore);
     verifyS3ConnectorName(featurestoreS3ConnectorDTO.getName(),featurestore, false);
     verifyS3ConnectorDescription(featurestoreS3ConnectorDTO.getDescription());
     verifyS3ConnectorBucket(featurestoreS3ConnectorDTO.getBucket());
-    verifyS3ConnectorAccessKey(featurestoreS3ConnectorDTO.getAccessKey());
-    verifyS3ConnectorSecretKey(featurestoreS3ConnectorDTO.getSecretKey());
+    
+    if(settings.isIAMRoleConfigured()){
+      verifySecretAndAccessKeysForIamRole(featurestoreS3ConnectorDTO);
+    }else{
+      verifyS3ConnectorAccessKey(featurestoreS3ConnectorDTO.getAccessKey());
+      verifyS3ConnectorSecretKey(featurestoreS3ConnectorDTO.getSecretKey());
+    }
+    if(featurestoreS3ConnectorDTO.getServerEncryptionAlgorithm() != null ||
+      !Strings.isNullOrEmpty(featurestoreS3ConnectorDTO.getServerEncryptionKey())){
```

Comment:
I would remove this check. the check that the encryption key is not empty happens only if the encryption key is required in the `verifyS3ConnectorServerEncryptionKey` We can remove it from here.

Reply:
also if remove it you can remove the check `verifyS3ConnectorServerEncryptionAlgorithm` as if you are in the IF you already know that the encryption algorithm is different than null.